### PR TITLE
planner: add store type to statement summary

### LINKED ISSUE #17278: Plan in statement summary tables is lack of tikv/tiflash info
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
mysql> explain select * from t;
+-----------------------+----------+-----------+---------------+--------------------------------+
| id                    | estRows  | task      | access object | operator info                  |
+-----------------------+----------+-----------+---------------+--------------------------------+
| TableReader_5         | 10000.00 | root      |               | data:TableFullScan_4           |
| └─TableFullScan_4     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo |
+-----------------------+----------+-----------+---------------+--------------------------------+
2 rows in set (0.01 sec)

mysql> select * from t;
Empty set (0.00 sec)

mysql> select plan from information_schema.statements_summary where digest_text='select * from t';
+----------------------------------------------------------------------------------------------------------------------+
| plan                                                                                                                 |
+----------------------------------------------------------------------------------------------------------------------+
| 	TableReader_5	root	10000	data:TableFullScan_4
	└─TableScan_4	cop 	10000	table:t, keep order:false, stats:pseudo     |
+----------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### 2. What did you expect to see? (Required)
Plan in `information_schema.statements_summary` should be the same with `explain`, which contains `tikv/tiflash` info.

### 3. Affected version (Required)

v4.0.0

### PR DESCRIPTION
**Pull request by pingyu**
<!-- Thank you for contributing to TiDB!

PR Title Format:
1. pkg [, pkg2, pkg3]: what's changed
2. *: what's changed

-->

### What problem does this PR solve?

Issue Number: close #17278 <!-- REMOVE this line if no issue to close -->

Problem Summary:

### What is changed and how it works?

What's Changed:
Add store type to `plan` in `information_schema.statements_summary`

How it Works:
Read store type from physical plan, and add to `plan` field.

### Related changes

### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->
- Manual test
```sql
mysql root@localhost:test> select * from t;                                                                                                                                                                                                 
0 rows in set
Time: 0.016s
mysql root@localhost:test>                                                                                                                                                                                                                  
mysql root@localhost:test> select plan from information_schema.statements_summary where digest_text='select * from t'; 
                                                                                                                                                                                                                                            
+------------------------------------------------------------------------------------+
| plan                                                                               |
+------------------------------------------------------------------------------------+
|     TableReader_5    root         10000    data:TableFullScan_4                    |
|     └─TableScan_4    cop[tikv]    10000    table:t, keep order:false, stats:pseudo |
+------------------------------------------------------------------------------------+
1 row in set
Time: 0.014s
mysql root@localhost:test>  
```

- Unit test

### Release note <!-- bugfixes or new feature need a release note -->

- Fix inconsistence of Plan between information_schema.statements_summary and explain, which contains tikv/tiflash info.
<!-- Please write a release note here to describe the change you made when it is released to the users of TiDB. If your PR doesn't involve any change to TiDB(like test enhancements, RFC proposals...), you can write `No release note`. -->

### CODE REVIEW

**Comment by breeswish** [comment]
Do we also need to fix slow log? @crazycs520

File: util/plancodec/codec.go [hunk 1]
```diff
@@ -316,6 +313,28 @@ func encodeID(planType string, id int) string {
 	return strconv.Itoa(planID) + idSeparator + strconv.Itoa(id)
 }
 
+func encodeTaskType(isRoot bool, storeType kv.StoreType) string {
```
Comment (hanfei1991) [inline util/plancodec/codec.go:316]: can we add unit tests for encode and decode TaskType ?

File: planner/core/encode.go [hunk 2]
```diff
@@ -57,8 +58,19 @@ func (pn *planEncoder) encodePlanTree(p PhysicalPlan) string {
...
+			storeType = copPlan.StoreType
+		case *PhysicalTableScan:
+			storeType = copPlan.StoreType
+		default:
+			storeType = kv.TiKV
+		}
+	}
 	actRows, analyzeInfo, memoryInfo, diskInfo := getRuntimeInfo(p.SCtx(), p)
-	plancodec.EncodePlanNode(depth, p.ID(), p.TP(), isRoot, p.statsInfo().RowCount, p.ExplainInfo(), actRows, analyzeInfo, memoryInfo, diskInfo, &pn.buf)
+	plancodec.EncodePlanNode(depth, p.ID(), p.TP(), isRoot, storeType, p.statsInfo().RowCount, p.ExplainInfo(), actRows, analyzeInfo, memoryInfo, diskInfo, &pn.buf)
```
Comment (hanfei1991) [inline planner/core/encode.go:73]: I think we can pass the encoded string of task deriectly.

File: util/plancodec/codec.go (hunk 1 above)
Comment (pingyu) [reply util/plancodec/codec.go]: Comment addressed. PTAL~ @hanfei1991

File: planner/core/encode.go (hunk 2 above)
Comment (pingyu) [reply planner/core/encode.go]: Comment addressed. PTAL~ @hanfei1991

**Comment by breeswish** [comment]
I'm interested in the root cause of this inconsistent. Is it caused by some information exists in memory (so that it serves `explain`), but was missing after serialization?

**Comment by breeswish** [comment]
@crazycs520 PTAL

**Comment by pingyu** [comment]
> LGTM
> 
> Please resolve the conflict.

Conflict resolved, PTAL, thanks~ @crazycs520 

p.s. The two failed CI tests seem to have nothing to do with this PR.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**